iso/controller/airootfs/etc/systemd/scripts/ceph-mon-init

```diff
@@ -54,33 +82,42 @@ chown ceph /etc/ceph/ceph.mon.keyring
 
 su -s /bin/bash -c "/usr/bin/ceph-mon --cluster ceph --mkfs -i ${CHOSTNAME} --monmap /etc/ceph/monmap --keyring /etc/ceph/ceph.mon.keyring" ceph
 
-export ETCDCTL_API=3
 
 touch /var/lib/ceph/mon/ceph-${CHOSTNAME}/done
 
-
 systemctl enable ceph.target ceph-mon.target
 systemctl enable ceph-mon@${CHOSTNAME}.service
 systemctl start ceph-mon@${CHOSTNAME}.service
 
-/usr/bin/ceph auth get-or-create mgr.${CHOSTNAME} mon 'allow profile mgr' osd 'allow *' mds 'allow *' > /var/lib/ceph/mgr/ceph-${CHOSTNAME}/keyring
-systemctl enable ceph-mgr@${CHOSTNAME}.service
-systemctl start ceph-mgr@${CHOSTNAME}.service
 
-/usr/bin/ceph --cluster ceph auth get-or-create client.kube mon 'allow r, allow command "osdblacklist"' osd 'allow rwx pool=kube'
+wait_for_unit 5 ceph-mon@${CHOSTNAME}
+if [ $? -ne 0 ] ; then
+	echo "FATAL: Ceph mon did not become available after 5 seconds"
+	exit 1
+fi
 
-sleep 3
+/usr/bin/ceph --cluster ceph auth get-or-create client.kube mon 'allow r, allow command "osdblacklist"' osd 'allow rwx pool=kube'
+/usr/bin/ceph --cluster ceph auth caps client.kube mon "profile rbd" osd "profile rbd pool=kube"
+/usr/bin/ceph auth get-or-create mgr.${CHOSTNAME} mon 'allow profile mgr' osd 'allow *' mds 'allow *' > /var/lib/ceph/mgr/ceph-${CHOSTNAME}/keyring
 
-etcd_cmd () {
-	        /usr/bin/etcdctl --endpoints=http://127.0.0.1:4279  "$@"
-}
+/usr/bin/ceph config-key set mgr/prometheus/server_addr ${OPEROS_CONTROLLER_IP}
 
+systemctl enable ceph-mgr@${CHOSTNAME}.service
+systemctl start ceph-mgr@${CHOSTNAME}.service
 
 cat /var/lib/ceph/mon/ceph-controller/keyring | etcd_cmd put "cluster/$OPEROS_INSTALL_ID/secret-ceph-mon-keyring"
 cat /etc/ceph/ceph.client.admin.keyring | etcd_cmd put "cluster/$OPEROS_INSTALL_ID/secret-ceph-client-admin-keyring"
 cat /etc/ceph/ceph.conf | etcd_cmd put "cluster/$OPEROS_INSTALL_ID/ceph-config"
 /usr/bin/ceph auth get client.kube | etcd_cmd put "cluster/$OPEROS_INSTALL_ID/secret-ceph-kube-keyring"
 
+wait_for_unit 5 ceph-mgr@${CHOSTNAME}
```

Comment:
I believe `systemctl start` already waits for the service to become active, so this shouldn't be necessary.

Reply:
you're right, going to have to think of something else or go back to the sleep hammer, this works because it  inserts a delay long enough between systemd starting the service and it actually becoming ready, might be able to check with commands but they usually have dozens of seconds timeouts